pkg/v20/resource/appmigration/create.go

```diff
@@ -168,6 +162,39 @@ func (r *Resource) newChartSpecsToMigrate() []key.ChartSpec {
 	return chartSpecsToMigrate
 }
 
+// waitForDeployedApp waits until the newly created app CR has a deployed helm
+// release and the chartconfig CR can be safely deleted.
+func (r *Resource) waitForDeployedApp(ctx context.Context, namespace, appName string) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waiting for deployed app CR %#q", appName))
+
+	o := func() error {
+		appCR, err := r.g8sClient.ApplicationV1alpha1().Apps(namespace).Get(appName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return microerror.Maskf(notFoundError, "app CR %#q", appName)
+		}
+
+		if appCR.Status.Release.Status != "DEPLOYED" {
+			return microerror.Maskf(notFoundError, "app CR %#q has status %#q", appName, appCR.Status.Release.Status)
```

Comment:
I think this is not the right error, `notFoundError`

Reply:
Changed to `notDeployedError` in #686. Still testing that PR.